BSP/STM32F413/Src/BSP_GPIO.c

```diff
@@ -0,0 +1,121 @@
+/* Copyright (c) 2021 UT Longhorn Racing Solar */
+
+#include "BSP_GPIO.h"
+#include "stm32f4xx.h"
+#include "Tasks.h"
+
+OS_SEM GPIO_Update_Sem4;
+
+/**
+ * @brief   Initializes a GPIO port
+ * @param   port to initialize
+ * @return  None
+ */ 
+void BSP_GPIO_Init(port_t port){

```

Comment:
See our recent discussion on this BSP for some idea of what could be potentially improved with the init function.

Reply:
Yeah, in the most recent code I still didn't change the init function, I'm just waiting to get some feedback from Rishi in regards to it because of possible conflicts with the simulator, but I copied the improvements you suggested as comments for now